src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/helper/builders/DSL.kt

```diff
@@ -19,9 +19,15 @@ import java.math.BigDecimal
 import java.time.LocalDate
 import java.time.LocalDateTime
 
+@Deprecated("To be replaced by TestDataFactory, coming soon")
 fun testData(repository: Repository, dsl: TestData.() -> Unit) = TestData(repository).apply(dsl)
 
-class TestData(private val repository: Repository) : TestDataDsl {
+@Component
+class TestDataFactory(private val repository: Repository, private val courseAllocationBuilderFactory: CourseAllocationBuilderFactory) {
+  fun build(dsl: TestData.() -> Unit) = TestData(repository, courseAllocationBuilderFactory).apply(dsl)
+}
```

Comment:
The long term goal of this and future changes is to use `TestDataFactory.build` to create all test data:
* each child element of `TestData` will have its `BuilderFactory` injected into this class
* each BuilderFactory will have its own dedicated Repository injected in by Spring
* each BuidlerFactory will also have any child builder factories injected in, each with their own dedicated Repository

Once everything is done Repository will have been split into dedicated repositories for each repo that are injected in by Spring.